eng/common/scripts/modules/ChangeLog-Operations.psm1

```diff
@@ -0,0 +1,127 @@
+            }
+            else
+            {
+               if ($releaseNotesEntry) {
+                  $releaseNotesEntry.ReleaseContent += $line
+               }
+            }
+        }
+    }
+    catch
+    {
+        Write-Host "Error parsing $ChangeLogLocation."
+        Write-Host $_.Exception.Message
+    }
+    return $releaseNotes
+}
+
+# Returns single ReleaseNotes object containing the ChangeLog for a particular version
+function Get-ReleaseNote {
+   param (
+      [Parameter(Mandatory = $true)]
+      [String]$ChangeLogLocation,
+      [Parameter(Mandatory = $true)]
+      [String]$VersionString
+   )
+
+   $releaseNotes = Get-ReleaseNotes -ChangeLogLocation $ChangeLogLocation
+
+   if ($releaseNotes.ContainsKey($VersionString)) 
+   {
+         return $releaseNotes[$VersionString]
+   }
+   Write-Error "Release Notes for the Specified version ${VersionString} was not found"
+   exit 1
+}
+
+#Returns the changelog for a particular version as string
+function Get-ReleaseNoteAsString {
+   param (
+      [Parameter(Mandatory = $true)]
+      [String]$ChangeLogLocation,
+      [Parameter(Mandatory = $true)]
+      [String]$VersionString
+   )
+
+   $releaseNotes = Get-ReleaseNote -ChangeLogLocation $ChangeLogLocation -VersionString $VersionString
+   [string]$releaseNotesTitle = $releaseNotes.ReleaseTitle
+   [string]$releaseNotesContent = $releaseNotes.ReleaseContent -Join [Environment]::NewLine
+   return $releaseNotesTitle, $releaseNotesContent -Join [Environment]::NewLine
+}
+
+function Confirm-ChangeLog {
+   param (
+      [Parameter(Mandatory = $true)]
+      [String]$ChangeLogLocation,
+      [Parameter(Mandatory = $true)]
+      [String]$VersionString,
+      [boolean]$ForRelease=$false
+   )
+
+   $ReleaseNotes = Get-ReleaseNote -ChangeLogLocation $ChangeLogLocation -VersionString $VersionString
+
+   if ([System.String]::IsNullOrEmpty($ReleaseNotes.ReleaseStatus))
+   {
+      Write-Host ("##[error]Changelog '{0}' has wrong release note title" -f $ChangeLogLocation)
+      Write-Host "##[info]Ensure the release date is included i.e. (yyyy-MM-dd) or (Unreleased) if not yet released"
+      exit 1
+   }
+
+   if ($ForRelease -eq $True)
+   {
+      $CurrentDate = Get-Date -Format "yyyy-MM-dd"
+      if ($ReleaseNotes.ReleaseStatus -ne "($CurrentDate)")
+      {
+         Write-Host ("##[warning]Incorrect Date: Please use the current date in the Changelog '{0}' before releasing the package" -f $ChangeLogLocation)
+         exit 1
+      }
+
+      if ([System.String]::IsNullOrWhiteSpace($ReleaseNotes.ReleaseContent))
+      {
+         Write-Host ("##[error]Empty Release Notes for '{0}' in '{1}'" -f $VersionString, $ChangeLogLocation)
+         Write-Host "##[info]Please ensure there is a release notes entry before releasing the package."
+         exit 1
+      }
+   }
+
+   Write-Host ($ReleaseNotes | Format-Table | Out-String)
+}
+ 
+Export-ModuleMember -Function 'Get-ReleaseNotes'
```

Comment:
We have some inconsistent naming in this file some places we call things changelog and other places we call them releasenote or releasenotes. Can I suggest the following function names:
```
Get-ChangeLogEntries
Get-ChangeLogEntry
Get-ChangeLogEntryAsString
Validate-ChangeLogEntry
```

Also for variables throughout I would rename `$releaseNote` with `$changeLogEntry` and `$releaseNotes` with `$changeLogEntries`

Reply:
Ok, Will do that.
I used Confirm to try and stay with PowerShell approved verbs. Validate produces and undesirable warning when installing the module.